fdbserver/DataDistribution.actor.cpp

```diff
@@ -1646,6 +1675,57 @@ struct DDTeamCollection : ReferenceCounted<DDTeamCollection> {
 		return std::pair<Reference<TCMachineTeamInfo>, int>(retMT, minNumProcessTeams);
 	}
 
+	// Find the machine team whose members are on the most number of machine teams, same logic as serverTeamRemover
+	std::pair<Reference<TCMachineTeamInfo>, int> getMachineTeamWithMostMachineTeams() {
+		Reference<TCMachineTeamInfo> retMT;
+		int maxNumMachineTeams = 0;
+		int targetMachineTeamNumPerMachine =
+		    (SERVER_KNOBS->DESIRED_TEAMS_PER_SERVER * (configuration.storageTeamSize + 1)) / 2;
+
+		for (auto& mt : machineTeams) {
+			// Do not remove a machine team if it cause 0 server team on a server
+			if (mt->existServerWithOnly1Team()) {
```

Comment:
replace this with if( representNumMachineTeams <= targetMachineTeamNumPerMachine ) continue;

Reply:
try removing existServerWithOnly1Team